packages/app/obojobo-repository/server/models/draft_summary.test.js

```diff
@@ -0,0 +1,516 @@
+			expect(db.one).toHaveBeenCalledWith(query, { id: 'mockDraftId' })
+			checkAgainstMockRawSummary(summary)
+		})
+	})
+
+	test('fetchById returns error when no match is found in the database', () => {
+		logger.error = jest.fn()
+
+		expect.hasAssertions()
+
+		db.one.mockRejectedValueOnce(new Error('not found in db'))
+
+		return DraftSummary.fetchById('mockDraftId').catch(err => {
+			expect(logger.error).toHaveBeenCalledWith('fetchById Error', 'not found in db')
+			expect(err).toBe('Error Loading DraftSummary by id')
+		})
+	})
+
+	test('fetchByUserId generates the correct query and returns a DraftSummary object', () => {
+		db.any = jest.fn()
+		db.any.mockResolvedValueOnce(mockRawDraftSummary)
+
+		const whereSQL = 'repository_map_user_to_draft.user_id = $[userId]'
+		const joinSQL = `JOIN repository_map_user_to_draft
+				ON repository_map_user_to_draft.draft_id = drafts.id`
+		const query = queryBuilder(whereSQL, joinSQL)
+
+		return DraftSummary.fetchByUserId(0).then(summary => {
+			expect(db.any).toHaveBeenCalledWith(query, { userId: 0 })
+			checkAgainstMockRawSummary(summary)
+		})
+	})
+
+	test('fetchAndJoinWhere catches database errors', () => {
+		expect.hasAssertions()
+
+		db.any.mockRejectedValueOnce(new Error('not found in db'))
+
+		const whereSQL = ''
+		const joinSQL = ''
+		const mockQueryValues = { id: 'mockDraftId' }
+
+		return DraftSummary.fetchAndJoinWhere(whereSQL, joinSQL, mockQueryValues).catch(err => {
+			expect(logger.error).toHaveBeenCalledWith(
+				'fetchAndJoinWhere Error',
+				'not found in db',
+				whereSQL,
+				joinSQL,
+				mockQueryValues
+			)
+			expect(err).toBe('Error loading DraftSummary by query')
+		})
+	})
+
+	test('fetchWhere catches database errors', () => {
+		expect.hasAssertions()
+
+		db.any.mockRejectedValueOnce(new Error('not found in db'))
+
+		const whereSQL = ''
+		const mockQueryValues = { id: 'mockDraftId' }
+
+		return DraftSummary.fetchWhere(whereSQL, mockQueryValues).catch(err => {
+			expect(logger.error).toHaveBeenCalledWith(
+				'fetchWhere Error',
+				'not found in db',
+				whereSQL,
+				mockQueryValues
+			)
+			expect(err).toBe('Error loading DraftSummary by query')
+		})
+	})
+
+	test('fetchAllDraftRevisions returns all versions of a draft when afterVersionId is null or missing', () => {
+		expect.hasAssertions()
+
+		db.any = jest.fn()
+		db.any.mockResolvedValueOnce(mockRawRevisionHistory)
+
+		const query = `
+			SELECT
+				drafts_content.id,
+				drafts_content.draft_id,
+				drafts_content.created_at,
+				drafts_content.user_id,
+				users.first_name,
+				users.last_name
+			FROM drafts_content
+			JOIN users
+				ON drafts_content.user_id = users.id
+			WHERE
+				drafts_content.draft_id = $[draftId]
+			ORDER BY
+				drafts_content.created_at DESC
+			LIMIT $[count];
+		`
+
+		return DraftSummary.fetchAllDraftRevisions('mockDraftId').then(history => {
+			expect(db.any).toHaveBeenCalledWith(query, {
```

Comment:
Quick idea, not sure, this maybe could use an 'expect string contains' against the important bits of the query?

Reply:
I think I might have tried that, but it ended up feeling a bit kludgy for queries where there was a lot to look for. I had kind of hoped there was some jest magic to ignore initial indentation, but that didn't pan out either.